### ISSUE TITLE
The test in the main driver to find the dylib directory doesn't work constantly.

### ISSUE THREAD

**Issue by joshuashaffer** [IBM/import-tracker#40]
## Describe the bug

The test in the main driver to find the dylib directory doesn't work constantly.

```sh
$ python -m import_tracker --name manage.py          
Traceback (most recent call last):         
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/jbs/three-ten-new/lib/python3.10/site-packages/import_tracker/__main__.py", line 41, in <module>
    _std_dylib_dir = os.path.realpath(os.path.dirname(cmath.__file__))
AttributeError: module 'cmath' has no attribute '__file__'. Did you mean: '__name__'?

```

## Platform

Please provide details about the environment you are using, including the following:

-   Interpreter version: Python 3.10.4 (main, Apr  9 2022, 21:27:52) [GCC 7.5.0] (Ubuntu 18.04.6 LTS)
-   Library version: 2.0.1

## Sample Code

Please include a minimal sample of the code that will (if possible) reproduce the bug in isolation
```python
In [1]: import cmath

In [2]: cmath.__file__
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [2], in <cell line: 1>()
----> 1 cmath.__file__

AttributeError: module 'cmath' has no attribute '__file__'

In [3]: 
```

## Expected behavior

The main driver to run.

## Observed behavior

The main driver crashes with the above stack trace.

**Comment by gabe-l-hart** [comment]
Hi @joshuashaffer, thanks for reporting this! I've tried to repro as closely as possible, but now luck. Here's what I did:

**bug.dockerfile**
```dockerfile
FROM ubuntu:18.04

ENV PATH=$PATH:/root/miniconda3/bin
RUN apt-get update && \
    apt-get install wget -y && \
    wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
    chmod +x Miniconda3-latest-Linux-x86_64.sh && \
    ./Miniconda3-latest-Linux-x86_64.sh -b && \
    conda create -n 3.10 python=3.10.4 -y
```

```sh
docker build . -f bug.dockerfile -t import_tracker_bug_env:latest
docker run --rm -it import_tracker_bug_env:latest

# Inside of docker image
conda init bash
bash
conda activate 3.10
python3 --version
# Python 3.10.4
python3 -c "import cmath; print(cmath.__file__)"
# /root/miniconda3/envs/3.10/lib/python3.10/lib-dynload/cmath.cpython-310-x86_64-linux-gnu.so
```

What kind of environment management are you using for your python environment? I could certainly imagine that different distributions of python 3.10 package these base shared objects differently.

**Comment by gabe-l-hart** [comment]
I also have `3.10` configured as a target environment for my test grid which is currently passing on `main`: https://github.com/IBM/import-tracker/runs/6600561205?check_suite_focus=true

**Comment by joshuashaffer** [comment]
I'm using a PPA. This behavior is consistent between different versions of
the interpreter and machines.

```
Package: libpython3.10-stdlib
Version: 3.10.4-1+bionic2
Priority: optional
Section: python
Source: python3.10
Maintainer: Matthias Klose ***@***.***>
Installed-Size: 8,079 kB
Depends: libpython3.10-minimal (= 3.10.4-1+bionic2), mime-support, tzdata,
libbz2-1.0, libc6 (>= 2.15), libdb5.3, libffi6 (>= 3.0.10~rc8), liblzma5
(>= 5.1.1alpha+20120614), libncursesw5 (>= 6), libreadline7 (>= 7.0~beta),
libsqlite3-0 (>= 3.14.0), libtinfo5 (>= 6)
Recommends: ca-certificates
Download-Size: 1,760 kB
APT-Manual-Installed: no
```

It's using the built-in BuiltinImporter to resolve cmath instead of the
SourceFileLoader which is why this is probably happening...
```python
In [14]: importlib.util.find_spec('os')
Out[14]: ModuleSpec(name='os',
loader=<_frozen_importlib_external.SourceFileLoader object at
0x7f4c16622ec0>, origin='/usr/lib/python3.10/os.py')

In [15]: importlib.util.find_spec('cmath')
Out[15]: ModuleSpec(name='cmath', loader=<class
'_frozen_importlib.BuiltinImporter'>, origin='built-in')
```

**Comment by gabe-l-hart** [comment]
Interesting! I'll look a bit deeper and see if I can repro and/or work around this. Thanks for the added details.

**Comment by clchiou** [comment]
I encountered the same issue. I guess this is due to cmath being built as a built-in module on Ubuntu, and so it [does not have](https://docs.python.org/3/using/configure.html#c-extensions) a `__file__` attribute. I think you may check whether your cmath module is built-in via checking whether "cmath" is in `sysconfig.get_config_var('MODBUILT_NAMES')`.

**Comment by gabe-l-hart** [comment]
@clchiou Great find! That definitely explains why it's behaving differently in a `conda` environment. I should be able to get a fix for this out in the next couple of days.

**Comment by gabe-l-hart** [comment]
Alright, I have a repro!

```dockerfile
FROM ubuntu:18.04

ENV PATH=$PATH:/root/miniconda3/bin
RUN true && \
    apt update && \
    apt install software-properties-common -y && \
    add-apt-repository ppa:deadsnakes/ppa -y && \
    apt update && \
    apt install python3.10 -y && \
    true

# This raises the error!
RUN python3.10 -c "import cmath; print(cmath.__file__)"
```

**Comment by gabe-l-hart** [comment]
This should be fixed in [2.0.3](https://github.com/IBM/import-tracker/releases/tag/2.0.3). Please let reopen this ticket if you're still experiencing this issue
